backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -267,30 +268,25 @@ def grant_dataset_bucket_key_policy(self):
             key_alias = f"alias/{self.target_bucket.KmsAlias}"
             kms_client = KmsClient(self.source_account_id, self.source_environment.region)
             kms_key_id = kms_client.get_key_id(key_alias)
-            existing_policy = kms_client.get_key_policy(kms_key_id)
-            target_requester_id = SessionHelper.get_role_id(self.target_account_id, self.target_requester_IAMRoleName)
-            if existing_policy and f'{target_requester_id}:*' not in existing_policy:
-                policy = json.loads(existing_policy)
-                policy["Statement"].append(
-                    {
-                        "Sid": f"{target_requester_id}",
-                        "Effect": "Allow",
-                        "Principal": {
-                            "AWS": "*"
-                        },
-                        "Action": "kms:Decrypt",
-                        "Resource": "*",
-                        "Condition": {
-                            "StringLike": {
-                                "aws:userId": f"{target_requester_id}:*"
-                            }
-                        }
-                    }
-                )
-                kms_client.put_key_policy(
-                    kms_key_id,
-                    json.dumps(policy)
-                )
+            existing_policy = json.loads(kms_client.get_key_policy(kms_key_id))
```

Comment:
What happens if we run in the exception in `get_key_policy` and it returns None? it will go into the else in line 280 and generate the default statement and then we will try to get existing_policy["Statement"] in line 285 and that I think it fails because of None["Statement"]

Reply:
We are assuming that the following kms key is always present:
```
{
            "Sid": "Enable IAM User Permissions",
            "Effect": "Allow",
            "Principal": {
                "AWS": [
                    "arn:aws:iam::xxxxx:root",
                    "arn:aws:iam::xxxxx:role/dataallPivotRole"
                ]
            },
            "Action": "kms:*",
            "Resource": "*"
        },
```
The reason for the assumption is that this is a pre-req step to using SSE-KMS encryption keys for the share. We have to add dataallPivotRole in statement Enable IAM User Permissions while importing a dataset.

Reply:
But this might cause issues when we use S3 encryption or someone deletes the policy by mistake. I have addressed this in next revision.